adr/ADR-001.md

```diff
@@ -0,0 +1,165 @@
+amount of blocks in a single year is `31,536,000`, but as the network gets more decentralised block-times will grow due
+to network latency, maybe raising to `5` seconds. This causes the amount of blocks in a single year to change, and hence
+the amount of inflation happening each block should grow too, to follow the shift in block-times. This causes inflation to
+be lower than the expected one. There's another case for which if block-times become faster, then the network will inflate
+more than the expected.
+
+Unfortunately, in order to re-adjust block provision a governance proposal is required to change the `x/mint.Params.BlocksPerYear`
+parameter. Relying on governance to actively adjust the target inflation through the `BlocksPerYear` param is infeasible long
+term as there can be block-time shifts as the proposal is ongoing.
+
+Considering Archway Network relies on inflation to provide part of the inflationary rewards to developers, the rewards stream 
+needs to be predictable and stable.
+
+## Decision
+
+We propose to create a new minting module which, instead of inflating a fixed amount each block, uses the elapsed time
+between blocks to decide the amount of staking tokens that need to be minted and directed to the modules.
+
+There's another difference between this module and the official cosmos-sdk's `x/mint`, ours should be able to correctly split
+inflation across multiple modules instead of a single one.
+
+### Params
+
+```protobuf
+syntax="proto3";
+
+import "google/protobuf/duration.proto";
+
+message Params {
+    // min_inflation defines the network's minimum inflation.
+    string min_inflation = 1; // sdk.Dec, percentage: 0 <= x <= 1
+    // max_inflation defines the network's maximum inflation.
+    string max_inflation = 2; // sdk.Dec, percentage: 0 <= x <= 1
+    // min_bonded defines the minimum wanted bond ratio (staked supply/total supply)
+    string min_bonded = 3; // sdk.Dec, percentage: 0 <= x <= 1
+    
+    // max_bonded defines the maximum wanted bond ratio (staked supply/total supply)
+    string max_bonded = 4; // sdk.Dec, percentage: 0 <= x <= 1
+    
+    // inflation_change defines how much the inflation should change if
+    // the bond ratio is not between the defined bands of min/max_bonded.
+    string inflation_change = 5; // sdk.Dec, percentage: 0 <= x <= 1.
+    
+    // max_block_duration defines the maximum duration of a block.
+    // this is important to have because there can be cases for which
+    // a block takes a lot longer than others, for example during upgrades.
+    // In these specific cases we don't want to have bursts of inflation
+    // in a single block.
+    google.protobuf.Duration max_block_duration = 6;
+    
+
+    // InflationRecipient defines the inflation recipients.
+    message InflationRecipient {
+      // recipient is the module receiving inflation, eg: x/staking, x/rewards.
+      string recipient = 1;
```

Comment:
Can only chain modules be inflation recipients or any general bech32 address?

Reply:
in theory only modules, if addresses (external owned accounts) want inflation probably a community spend proposal is better than using minting.